### PR TITLE
tests: testify the autoscaling/compatibility/dashboard/pdbackup tests

### PR DESCRIPTION
**Pull request by JmPotato**
Signed-off-by: JmPotato <ghzpotato@gmail.com>

### What problem does this PR solve?

Issue Number: ref #4813.

### What is changed and how does it work?

```commit-message
Testify the autoscaling/compatibility/dashboard/pdbackup tests.
```

### Check List

Tests

- Unit test
- Integration test

### Release note

```release-note
None.
```

### CODE REVIEW

File: tests/dashboard/race_test.go [hunk 1]
```diff
@@ -28,35 +28,27 @@ import (
...
+	re.NoError(cluster.RunInitialServers())
 	cluster.WaitLeader()
 
 	time.Sleep(60 * time.Millisecond)
 	cancel()
+
+	server.EnableZap = false
+	dashboard.SetCheckInterval(time.Second)
+	tests.WaitLeaderReturnDelay = 20 * time.Millisecond
+	tests.WaitLeaderCheckInterval = 500 * time.Millisecond
```
Comment (rleungx) [inline tests/dashboard/race_test.go:53]: use a defer?

File: tests/autoscaling/autoscaling_test.go [hunk 2]
```diff
@@ -102,7 +95,7 @@ func (s *apiTestSuite) TestAPI(c *C) {
     ]
 }`)
 	resp, err := http.Post(leaderServer.GetAddr()+"/autoscaling", "application/json", bytes.NewBuffer(jsonStr))
-	c.Assert(err, IsNil)
+	re.NoError(err)
 	defer resp.Body.Close()
-	c.Assert(resp.StatusCode, Equals, 200)
+	re.Equal(200, resp.StatusCode)
```
[the remark this answers is not in the document]
Comment (JmPotato) [reply tests/autoscaling/autoscaling_test.go:100]: `require.HTTPSuccess` needs an HTTP handler as the first argument.

**Review by rleungx — APPROVED** [review]

**Review by AndreMouche — APPROVED** [review]
LGTM

**Pull request merged**